tests/TestCases/TemporalExpressions/TEDayOfMonthTest.php

```diff
@@ -135,4 +135,57 @@ public function testCorrectDateAcrossYearsWithCorrectFrequencyReturnsTrue()
 
         $this->assertTrue($result);
     }
+
+    public function testIgnoredDateInPatternReturnsFalse()
+    {
+        $pattern = TEDayofMonth::build(new Carbon('2021-01-01'), 1);
+        $testDate = new Carbon('2021-06-01');
+
+        $this->assertTrue($pattern->includes($testDate));
+
+        $pattern->setIgnoreDates([$testDate]);
+
+        $this->assertFalse($pattern->includes($testDate));
+    }
+
+    public function testNextSelectsCorrectDate()
+    {
+        $pattern = TEDayOfMonth::build(new Carbon('2021-01-01'), 1);
+        $next = $pattern->next();
+
+        $this->assertEquals(
+            $next->toDateString(),
+            '2021-02-01'
+        );
+    }
+
+    public function testNextWithFrequencySelectsCorrectDate()
+    {
+        $pattern = TEDayOfMonth::build(new Carbon('2021-01-01'), 1);
+        $pattern->setFrequency(3);
+        $next = $pattern->next();
+
+        $this->assertEquals(
+            $next->toDateString(),
+            '2021-04-01'
+        );
+    }
+
+    public function testNextWithInvalidCurrentDateSelectsCorrectDate()
+    {
+        //TODO: Implement with seek() to invalid date, should select next valid date
+        $this->assertTrue(false);
+    }
+
+    public function testNextDateAfterPatternEndReturnsNull()
+    {
+        $pattern = TEDayOfMonth::build(new Carbon('2021-01-01'), 1)
+            ->setEndDate(new Carbon('2021-06-01'));
+
+        while ($pattern->valid()) {
+            $pattern->next();
+        }
```

Comment:
I see a few problems here. Given the way you have `next()` implemented right now, and the only reason this doesn't result in an infinite loop is actually what I mentioned earlier about `addMonths()` actually modifying `$this->current` directly. Otherwise, because you never actually say `$this->current = $next` unless `$next` is in the pattern, you'd get an infinite loop stuck on the last date of the pattern.

In general I think unit tests should set the conditions a little more explicitly. When possible, avoid loops for situations because it obscures what's actually happening, and you can miss problems like the one I just pointed out. It would be better if you explicitly called `$pattern->next()` 6 times and checked the result after each one.